src/apple/deps/mod.rs

```diff
@@ -33,20 +80,25 @@ pub enum Error {
     PromptFailed(#[from] std::io::Error),
     #[error(transparent)]
     VersionLookupFailed(#[from] system_profile::Error),
+    #[error("Failed to update package `{package}`")]
+    PackageNotUpdated { package: &'static str },
+    #[error("Failed to list installed gems: {0}")]
+    GemListFailed(#[from] bossy::Error),
+    #[error("Regex match failed for output of `gem list`")]
+    RegexMatchFailed,
 }
 
-pub fn install(package: &'static str, reinstall_deps: opts::ReinstallDeps) -> Result<bool, Error> {
-    let found = util::command_present(package)
-        .map_err(|source| Error::PresenceCheckFailed { package, source })?;
-    log::info!("package `{}` present: {}", package, found);
-    if !found || reinstall_deps.yes() {
-        println!("Installing `{}`...", package);
-        // reinstall works even if it's not installed yet, and will upgrade
-        // if it's already installed!
-        bossy::Command::impure_parse("brew reinstall")
-            .with_arg(package)
-            .run_and_wait()
-            .map_err(|source| Error::InstallFailed { package, source })?;
+pub fn install(
```

Comment:
This and other functions that take `&PackageSpec` could also be methods on `PackageSpec`